### DISCUSSION TITLE
How to create getter for a class?

### DISCUSSION THREAD

**Discussion by enoch-aik** [serverpod/serverpod#1183] [Q&A] [answered]
I am trying out Serverpod for the first time today and it has been awesome so far!. I want to create a class that has a getter in it, how can this be done in protocol/*.yaml 

Example, I am trying to create a class like this:

class Guest {
  final String name;
  final DateTime? checkInTime;
  final DateTime? checkOutTime;

bool get yetToCheckout => checkOutTime?.isAfter(DateTime.now()) ?? true;

...
}

How do I completely define this class in .yaml and also assign the logic for yetToCheckout like in the sample class above?

class: Guest
table: guest
fields: 
  name: String
  checkInTime: DateTime?
  checkOutTime: DateTime?
  yetToCheckout: bool

**Comment by netkeojas** [comment, marked as answer]
Hi, if you want to have a custom getter , i.e having your own logic, I don't this so you can do it in the protocol.yaml, what you can do is as follows: 
1. generate your class and table , serverpod generate.
2. once you generate, serverpod creates a class for you with the name Guest, now this class will have all the fields which can be accessed.

now you have 2 option
1. edit this class as per your requirement
2. write an endpoint file where you can access the data from the generated class and write your logic here.
